c/tests/test_trees.c

```diff
@@ -1157,6 +1157,134 @@ test_simplest_discrete_time(void)
     tsk_table_collection_free(&tables);
 }
 
+static void
+test_simplest_min_time(void)
+{
+    int ret;
+    tsk_treeseq_t ts;
+    tsk_table_collection_t tables;
+    const char *nodes = "is_sample  individual  time\n"
+                        "1  0   0.1\n"
+                        "1  0   0.1\n"
+                        "1  0   0.1\n"
+                        "0  -1   1.0\n"
+                        "0  -1   2.0";
+    const char *edges = "left   right   parent  child\n"
+                        "0  2   3   0\n"
+                        "0  2   3   1\n"
+                        "0  2   4   2\n"
+                        "0  2   4   3";
+    const char *sites = "position   ancestral_state\n"
+                        "0.0  0\n"
+                        "1.0  0";
+    const char *mutations = "site   node    derived_state   parent  time\n"
+                            "0  3   1   -1  1.5\n"
+                            "1  2   1   -1  0.5";
+
+    tsk_treeseq_from_text(&ts, 1, nodes, edges, NULL, sites, mutations, NULL, NULL, 0);
+    CU_ASSERT_EQUAL(tsk_treeseq_get_min_time(&ts), 0.1);
+
+    ret = tsk_table_collection_copy(ts.tables, &tables, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    tsk_treeseq_free(&ts);
+
+    ret = tsk_treeseq_init(&ts, &tables, TSK_TS_INIT_BUILD_INDEXES);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_EQUAL(tsk_treeseq_get_min_time(&ts), 0.1);
+    tsk_treeseq_free(&ts);
+
+    /* Setting mutation times to unknown should have no effect on min time. */
+    tables.mutations.time[0] = TSK_UNKNOWN_TIME;
+    tables.mutations.time[1] = TSK_UNKNOWN_TIME;
+    ret = tsk_treeseq_init(&ts, &tables, TSK_TS_INIT_BUILD_INDEXES);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_EQUAL(tsk_treeseq_get_min_time(&ts), 0.1);
+    tsk_treeseq_free(&ts);
+    tables.mutations.time[0] = 1.5;
+    tables.mutations.time[1] = 0.5;
+
+    /* A bad mutation younger than the youngest node. */
+    tables.mutations.time[1] = 0.0;
```

Comment:
Don't need to bother with this - the values within the tree sequence are undefined if there's an error

Reply:
Well I meant there's no need to test this case at all - if it's an error then the ``min_time`` is undefined.

Reply:
Ah, I didn't see these comments!